x-pack/plugins/fields_metadata/README.md

```diff
@@ -0,0 +1,168 @@
+
+#### Start Contract
+
+The start contract exposes a `FieldsMetadataClient` instance, which offers the following methods:
+- `getByName(name: string, params? {integration: string, dataset?: string})`: Retrieves a single `FieldMetadata` instance by name.
+
+```ts
+const timestampField = await client.getByName('@timestamp')
+/*
+{
+  dashed_name: 'timestamp',
+  type: 'date',
+  ...
+}
+*/
+```
+
+- `find(params?: {fieldNames: string[], integration: string, dataset?: string})`: Retrieves a record of matching `FieldMetadata` instances based on the query parameters.
+
+**Parameters**
+| Name | Type | Example | Optional |
+|---|---|---|---|
+| fieldNames | <EcsFieldName \| string>[] | ['@timestamp', 'onepassword.client.platform_version'] | ✅ |
+| integration | string | 1password | ✅ |
+| dataset | string | 1password.item_usages | ✅ |
+
+```ts
+const fields = await client.find({
+  fieldNames: ['@timestamp', 'onepassword.client.platform_version'], 
+  integration: '1password'
+})
+/*
+{
+  '@timestamp': {
+    dashed_name: 'timestamp',
+    type: 'date',
+    ...
+  },
+  'onepassword.client.platform_version': {
+    name: 'platform_version',
+    type: 'keyword',
+    ...
+  },
+}
+*/
+```
+
+> N.B. Passing the `dataset` name parameter to `.find` helps narrowing the scope of the integration assets that need to be fetched, increasing the performance of the request. Using only the `integration` parameter should achieve the same result, but is recommended always passing the `dataset` as well if known or unless the required fields come from different datasets of the same integration.
+
+> N.B. In case the `fieldNames` parameter is not passed to `.find`, the result will give the whole list of ECS fields by default. This should be avoided as much as possible, although it helps covering cases where we might need the whole ECS fields list.
+
+#### Source Repositories
+
+The `FieldsMetadataClient` relies on source repositories to fetch field metadata. Currently, there are two repository sources:
+- `EcsFieldsRepository`: Fetches static ECS field metadata.
+- `IntegrationFieldsRepository`: Fetches fields from an integration package from the Elastic Package Registry (EPR). This requires the `fleet` plugin to be enabled to access the registered fields extractor.
+
+To improve performance, a caching layer is applied to the results retrieved from external sources, minimizing latency and enhancing efficiency.
+
+### Fields Metadata API
+
+A REST API endpoint is exposed to facilitate the retrieval of field metadata:
+
+- `GET /internal/fields_metadata/find`: Supports query parameters to filter and find field metadata, optimizing the payload served to the client.
+
+**Parameters**
+| Name | Type | Example | Optional |
+|---|---|---|---|
+| fieldNames | <EcsFieldName \| string>[] | ['@timestamp', 'onepassword.client.platform_version'] | ✅ |
+| attributes | FieldAttribute[] | ['type', 'description', 'name'] | ✅ |
```

Comment:
Its unclear to me that there's much benefit to getting a subset of attributes instead of simply getting the whole set of attributes. 

I'm familiar with the use case of showing field descriptions to users. Are there other use cases?

Reply:
The main benefit is reduced response payload when consumption happens on the client side.

For instance, in the flyout fields we only need a couple of attributes (flat_name, short description and type), while returning the whole set of attributes would almost double the payload size for information not used at all. 

Do you have any concerns about having a filtered subset of attributes based on use cases against the whole field object?